spacy/tokens/doc.pyx

```diff
@@ -1462,6 +1462,146 @@ cdef class Doc:
                 remove_label_if_necessary(attributes[i])
                 retokenizer.merge(span, attributes[i])
 
+    def from_json(self, doc_json):
+        """Convert a JSON document generated by Doc.to_json() to a Doc.
+
+        vocab (Vocab): A storage container for lexical types.
+        RETURNS (Doc): A doc instance corresponding to the specified JSON representation.
+        """
+
+        ### Token-level properties ###
+
+        words = []
+        # Map annotation type IDs to their string equivalents.
+        annot_types = {t: self.vocab.strings[t].lower() for t in (POS, HEAD, DEP, LEMMA, TAG, MORPH)}
+        token_annotations_reqs = {
+            annot_type_id: annot_type_id if annot_type_id != HEAD else DEP for annot_type_id in annot_types
+        }
+        token_annotations = {annot_type_id: None for annot_type_id in annot_types}
```

Comment:
Can you refactor this so that you only have keys for the annotation that's present instead of using `None`?

I think this makes the checks below clearer and doesn't potentially confuse `None` vs. `[]` which will both evaluate to `False` but not mean the same thing.